src/runtime/runtime_mimxrt1062_clock.go

```diff
@@ -0,0 +1,1242 @@
+			^((nxp.CCM_ANALOG_PFD_528_PFD0_CLKGATE_Msk | nxp.CCM_ANALOG_PFD_528_PFD0_FRAC_Msk) << (8 * uint32(i)))
+		frac := (val << nxp.CCM_ANALOG_PFD_528_PFD0_FRAC_Pos) & nxp.CCM_ANALOG_PFD_528_PFD0_FRAC_Msk
+		// disable the clock output first
+		nxp.CCM_ANALOG.PFD_528.Set(pfd528 | (nxp.CCM_ANALOG_PFD_528_PFD0_CLKGATE_Msk << (8 * uint32(i))))
+		// set the new value and enable output
+		nxp.CCM_ANALOG.PFD_528.Set(pfd528 | (frac << (8 * uint32(i))))
+	}
+}
+
+func setUsb1Pfd(value ...uint32) {
+	for i, val := range value {
+		pfd480 := nxp.CCM_ANALOG.PFD_480.Get() &
+			^((nxp.CCM_ANALOG_PFD_480_PFD0_CLKGATE_Msk | nxp.CCM_ANALOG_PFD_480_PFD0_FRAC_Msk) << (8 * uint32(i)))
+		frac := (val << nxp.CCM_ANALOG_PFD_480_PFD0_FRAC_Pos) & nxp.CCM_ANALOG_PFD_480_PFD0_FRAC_Msk
+		// disable the clock output first
+		nxp.CCM_ANALOG.PFD_480.Set(pfd480 | (nxp.CCM_ANALOG_PFD_480_PFD0_CLKGATE_Msk << (8 * uint32(i))))
+		// set the new value and enable output
+		nxp.CCM_ANALOG.PFD_480.Set(pfd480 | (frac << (8 * uint32(i))))
+	}
+}
+
+func (clk clock) isPllEnabled() bool {
+	const ccmAnalogBase = 0x400d8000
+	addr := ccmAnalogBase + ((uint32(clk) >> 16) & 0xFFF)
+	pos := uint32(1 << (uint32(clk) & 0x1F))
+	return ((*volatile.Register32)(unsafe.Pointer(uintptr(addr)))).HasBits(pos)
+}
+
+func (clk clock) isPllBypassed() bool {
+	const ccmAnalogBase = 0x400d8000
+	addr := ccmAnalogBase + ((uint32(clk) >> 16) & 0xFFF)
+	pos := uint32(1 << pllBypassPos)
+	return ((*volatile.Register32)(unsafe.Pointer(uintptr(addr)))).HasBits(pos)
+}
+
+func (clk clock) getBypassFreq() uint32 {
+	const ccmAnalogBase = 0x400d8000
+	addr := ccmAnalogBase + ((uint32(clk) >> 16) & 0xFFF)
+	src := (((*volatile.Register32)(unsafe.Pointer(uintptr(addr)))).Get() &
+		pllBypassClkSrcMsk) >> pllBypassClkSrcPos
+	if src == uint32(pllSrc24M) {
+		return getOscFreq()
+	}
+	return 0
+}
+
+func (clk clock) bypass(bypass bool) {
+	const ccmAnalogBase = 0x400d8000
+	if bypass {
+		addr := ccmAnalogBase + ((uint32(clk) >> 16) & 0xFFF) + 4
+		((*volatile.Register32)(unsafe.Pointer(uintptr(addr)))).Set(1 << pllBypassPos)
+	} else {
+		addr := ccmAnalogBase + ((uint32(clk) >> 16) & 0xFFF) + 8
+		((*volatile.Register32)(unsafe.Pointer(uintptr(addr)))).Set(1 << pllBypassPos)
+	}
+}
+
+const (
+	oscRc   = 0 // On chip OSC
+	oscXtal = 1 // 24M Xtal OSC
+)
+
+type gate uint8
+
+const (
+	gateClkNotNeeded     gate = 0 // Clock is off during all modes
+	gateClkNeededRun     gate = 1 // Clock is on in run mode, but off in WAIT and STOP modes
+	gateClkNeededRunWait gate = 3 // Clock is on during all modes, except STOP mode
+)
+
+type clockMode uint8
+
+const (
+	modeClkRun  clockMode = 0 // Remain in run mode
+	modeClkWait clockMode = 1 // Transfer to wait mode
+	modeClkStop clockMode = 2 // Transfer to stop mode
+)
+
+func (m clockMode) set() {
+	nxp.CCM.CLPCR.Set((nxp.CCM.CLPCR.Get() & ^uint32(nxp.CCM_CLPCR_LPM_Msk)) |
+		((uint32(m) << nxp.CCM_CLPCR_LPM_Pos) & nxp.CCM_CLPCR_LPM_Msk))
+}
+
+const (
```

Comment:
`device` probably.

Reply:
Same as previous (moved to new file `src/device/nxp/clock_mimxrt1062.go`)